tests/components/yessssms/test_notify.py

```diff
@@ -151,210 +149,219 @@ async def test_connection_error_on_init(hass, caplog, valid_settings, connection
-        mock.register_uri(
-            requests_mock.POST,
-            # pylint: disable=protected-access
-            self.yessssms.yesss._login_url,
-            status_code=200,
-            text="BlaBlaBla<strong>Login nicht erfolgreichBlaBla",
-        )
-
-        message = "Testing YesssSMS platform :)"
-
-        with self.assertLogs("homeassistant.components.yessssms.notify", level="ERROR"):
-            self.yessssms.send_message(message)
-        self.assertTrue(mock.called)
-        self.assertEqual(mock.call_count, 1)
-
-    def test_empty_message_error(self):
-        """Test for an empty SMS message error."""
-        message = ""
-        with self.assertLogs("homeassistant.components.yessssms.notify", level="ERROR"):
-            self.yessssms.send_message(message)
-
-    @requests_mock.Mocker()
-    def test_error_account_suspended(self, mock):
-        """Test login that fails after multiple attempts."""
-        mock.register_uri(
-            "POST",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._login_url,
-            status_code=200,
-            text="BlaBlaBla<strong>Login nicht erfolgreichBlaBla",
-        )
-
-        message = "Testing YesssSMS platform :)"
-
-        with self.assertLogs("homeassistant.components.yessssms.notify", level="ERROR"):
-            self.yessssms.send_message(message)
-        self.assertTrue(mock.called)
-        self.assertEqual(mock.call_count, 1)
-
-        mock.register_uri(
-            "POST",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._login_url,
-            status_code=200,
-            text="Wegen 3 ungültigen Login-Versuchen ist Ihr Account für "
-            "eine Stunde gesperrt.",
-        )
-
-        message = "Testing YesssSMS platform :)"
-
-        with self.assertLogs("homeassistant.components.yessssms.notify", level="ERROR"):
-            self.yessssms.send_message(message)
-        self.assertTrue(mock.called)
-        self.assertEqual(mock.call_count, 2)
-
-    def test_error_account_suspended_2(self):
-        """Test login that fails after multiple attempts."""
-        message = "Testing YesssSMS platform :)"
+@pytest.fixture(name="yessssms")
+def yessssms_init():
+    """Set up things to be run when tests are started."""
+    login = "06641234567"
+    passwd = "testpasswd"
+    recipient = "06501234567"
+    client = yessssms.YesssSMS(login, passwd)
+    return yessssms.YesssSMSNotificationService(client, recipient)
+
+
+async def test_login_error(yessssms, requests_mock, caplog):
+    """Test login that fails."""
+    requests_mock.post(
+        # pylint: disable=protected-access
+        yessssms.yesss._login_url,
+        status_code=200,
+        text="BlaBlaBla<strong>Login nicht erfolgreichBlaBla",
+    )
+
+    message = "Testing YesssSMS platform :)"
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+    assert requests_mock.called is True
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)